core/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/SameShardAllocationDecider.java

```diff
@@ -62,7 +62,8 @@ public Decision canAllocate(ShardRouting shardRouting, RoutingNode node, Routing
         for (ShardRouting assignedShard : assignedShards) {
             if (node.nodeId().equals(assignedShard.currentNodeId())) {
                 return allocation.decision(Decision.NO, NAME,
-                        "the shard cannot be allocated on the same node id [%s] on which it already exists", node.nodeId());
+                        "the shard cannot be allocated to the same node id [%s] on which a copy of the shard already exists",
```

Comment:
instead of saying "allocated to the same node *id*" it should just say "to the same  *node*". I wonder if we should provide more info here as well, maybe we could put the whole shard routing object instead of the node id. I also wonder whether it makes to distinguish between the same shardrouting object and a different one (i.e. are we trying to allocate to the node on which WE are already allocated or are trying to allocate to a node where another copy of the shard is already allocated)